hermes-common/src/main/java/pl/allegro/tech/hermes/domain/topic/TopicRepository.java

```diff
@@ -29,4 +29,6 @@
     List<Topic> getTopicsDetails(Collection<TopicName> topicNames);
 
     boolean isSubscribingRestricted(TopicName topicName);
+
+    void ensureTopicIsEmpty(TopicName topicName);
```

Comment:
Since you are making it public, maybe it's a good time to rename it to `ensureTopicHasNoSubscriptions`?
If it wasn't for the name of the pull request I would think it checks if a topic has published messages.